dulwich/porcelain.py

```diff
@@ -2098,6 +2099,123 @@ def checkout_branch(repo, target: Union[bytes, str], force: bool = False) -> Non
                 dir_path = os.path.dirname(dir_path)
 
 
+# --- Sparse Checkout Commands ---
+
+
+def sparse_checkout(repo, patterns=None, force=False):
+    """Perform a sparse checkout by updating the index and working tree
+    based on inclusion/exclusion patterns, using skip-worktree bits in
+    the index's extended_flags field (as supported by patched Dulwich).
```

Comment:
not sure what you mean by patched Dulwich?

Reply:
sorry, that was intermediate notes to self about the separate work with the `extended_flags` in the index, rewritten more clearly and in proper D205 format :ballot_box_with_check: